controllers/datadogagent/dependencies/store.go

```diff
@@ -0,0 +1,227 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package dependencies
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-operator/pkg/equality"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// OperatorStoreLabelKey used to identified which resource is managed by the store.
+	OperatorStoreLabelKey = "operator.datadoghq.com/managed-by-store"
+	// OperatorGenHashAnnotationKey annotation key used on a Resource in order to compare 2 resources without know the spec struct.
+	OperatorGenHashAnnotationKey = "operator.datadoghq.com/gen-hash"
```

Comment:
Not used, could be easily implemented in the store, no?

Reply:
it already the case and used here:
* https://github.com/DataDog/datadog-operator/pull/412/files#diff-48d116016241a7d62a8614e21fc9b23f37962ca9d1bb54fa2ee5abd447709ed3R74
* https://github.com/DataDog/datadog-operator/pull/412/files#diff-48d116016241a7d62a8614e21fc9b23f37962ca9d1bb54fa2ee5abd447709ed3R146